torchvision/utils.py

```diff
@@ -191,34 +192,42 @@ def draw_bounding_boxes(
     elif image.size(0) not in {1, 3}:
         raise ValueError("Only grayscale and RGB images are supported")
 
+    if labels:
+        if len(labels) != boxes.size(0):
+            raise ValueError("Number of boxes and labels mismatch. Please specify labels for each box.")
```

Comment:
Nit: we could indicate the number of boxes and labels to give a more actionable error message:

```suggestion
            raise ValueError(f"Number of boxes ({boxes.size(0)}) and labels ({len(labels)}) mismatch. Please specify labels for each box.")
```